src/senaite/core/upgrade/v02_06_000.py

```diff
@@ -93,26 +139,64 @@ def migrate_departments_to_dx(tool):
     tool.runImportStepFromProfile(profile, "typeinfo")
     tool.runImportStepFromProfile(profile, "workflow")
 
-    old_id = "bika_departments"
-    new_id = "departments"
+    # get the old container
+    origin = api.get_setup().get("bika_sampleconditions")
+    if not origin:
+        # old container is already gone
+        return
 
-    setup = api.get_senaite_setup()
-    bikasetup = api.get_setup()
+    # get the destination container
+    destination = get_setup_folder("sampleconditions")
 
-    old = bikasetup.get(old_id)
-    new = setup.get(new_id)
+    # un-catalog the old container
+    uncatalog_object(origin)
 
-    if not new:
-        portal = api.get_portal()
-        add_senaite_setup_items(portal)
-        new = setup.get(new_id)
+    # Mapping from schema field name to a tuple of
+    # (accessor, target field name, default value)
+    schema_mapping = {
+        "title": ("Title", "title", ""),
+        "description": ("Description", "description", ""),
+    }
+
+    # migrate the contents from the old AT container to the new one
+    migrate_to_dx("SampleCondition", origin, destination, schema_mapping)
+
+    # copy snapshots for the container
+    copy_snapshots(origin, destination)
+
+    # remove old AT folder
+    if len(origin) == 0:
+        delete_object(origin)
```

Comment:
Why this test for `len(origin)`?

Reply:
For e.g. SampleConditions we donknow the folder will contain objects from SanpleCondition type. We cannot say the same for other folders we might migrate objects from in future (like e.g. Client when migrating SampleType objects)

Reply:
Ah, now I got it. You are deleting it explicitly in `migrate_to_dx`:
```
        # Migrate the contents from AT to DX
        migrator = getMultiAdapter(
            (src, target), interface=IContentMigrator)
        migrator.migrate(schema_mapping, delete_src=False)

        # Delete the old AT object
        delete_object(src)
```

Please change the code above and pass in `delete_src` to the migrator instead.